### PR TITLE
Update and standardize implementation of RADIUSS packages

### PR DESCRIPTION
**Pull request by adrienbernede**
This PR :
- migrates CARE and Caliper to CachedCMakePackage, closing the gap with implementations found in llnl/radiuss-spack-configs.
- adds latest releases.
- improves coherency in version constraints across RADIUSS packages.
- adds chai and care in radiuss stack for CI
- adds myself as maintainer in modified packages, fixes license when necessary, and other small changes.

### CODE REVIEW

**Comment by adrienbernede** [comment]
Forced push to rebase on latest snapshot release so that CI can run.

File: var/spack/repos/builtin/packages/umpire/package.py
Comment (adayton1) [inline var/spack/repos/builtin/packages/umpire/package.py:25]: Keep this extra line between the license and all the versions.

**Review by adayton1 — APPROVED** [review]
LGTM

File: var/spack/repos/builtin/packages/care/package.py
Comment (adayton1) [inline var/spack/repos/builtin/packages/care/package.py]: I don't think I understood this before, but I don't think we need the submodules.

Comment (adrienbernede) [reply var/spack/repos/builtin/packages/care/package.py]: OK, will have to test that change in our CI.
Would you know whether they were needed at some point ?

**Comment by adrienbernede** [comment]
@adayton1 would you know why CHAI is not finding RAJA anymore ?
https://gitlab.spack.io/spack/spack/-/jobs/12157797

File: var/spack/repos/builtin/packages/care/package.py
Comment (adayton1) [reply var/spack/repos/builtin/packages/care/package.py]: I'm not sure they were ever needed.

**Comment by adayton1** [comment]
> @adayton1 would you know why CHAI is not finding RAJA anymore ? https://gitlab.spack.io/spack/spack/-/jobs/12157797

It looks like RAJA was never installed in the job. So it seems that it recognized that the RAJA variant was requested, but never picked up the dependency. Maybe there needs to be a simple depends_on("raja") line?

**Review by v-dobrev — DISMISSED** [review]
The changes to the `mfem` package look good to me.

File: var/spack/repos/builtin/packages/umpire/package.py
Comment (adrienbernede) [reply var/spack/repos/builtin/packages/umpire/package.py:25]: Done

**Comment by adrienbernede** [comment]
@adayton1 The failure with CHAI is due to CHAI build system still looking for RAJA while the variant is set to `~raja`...

**Comment by adrienbernede** [comment]
@cyrush I’d like your validation regarding the requirement on the RAJA version in Ascent.

**Comment by cyrush** [comment]
@adrienbernede  here are the details for Ascent 0.9.3:

* conduit@0.9.1 (newer should be fine)
* vtk-m@2.1.0 (requires [patch](https://github.com/Alpine-DAV/ascent/blob/0aef6cffd522be7419651e6adf586f9a553297d0/scripts/build_ascent/2024_05_03_vtkm-mr3215-ext-geom-fix.patch) )
* raja@2024.02.1
* umpire@2024.02.1
* camp@2024.02.1
* kokkos@3.7.02

These are from:

https://github.com/Alpine-DAV/ascent/blob/develop/CHANGELOG.md#093---released-2024-05-11

File: var/spack/repos/builtin/packages/chai/package.py
Comment (adayton1) [inline var/spack/repos/builtin/packages/chai/package.py:281]: Did this fix the CHAI failures you were seeing?

Comment (adrienbernede) [reply var/spack/repos/builtin/packages/chai/package.py:281]: Yes

**Comment by adrienbernede** [comment]
@cyrush you are listing the "Preferred dependency versions for ascent@0.9.3".
Likely, it means that anything newer is not guaranteed to be supported.
In which case I should apply to Umpire the same version constraint I added for RAJA.
Is that what you are suggesting ?

Note: I could also write a constraint on Camp version through Umpire and/or RAJA, but I think we should assume that the Umpire and RAJA packages will do a good job at enforcing an appropriate version. If Ascent adds specific requirements to the Camp version, then we should express those in the Ascent package.

**Comment by cyrush** [comment]
@adrienbernede  That shows what we have tested & we can't really guarantee Ascent will work with over versions.  If we had testing in spack we could try to loosen, but I don't think we have good testing to try an experiment with any version.

**Comment by adrienbernede** [comment]
@cyrush Understood. Then I think I should also have Ascent@0.9.3 explicitly require vtk-m@2.1.0, which is not the case currently.

**Comment by adrienbernede** [comment]
Note: ascent being built with vtk-m 2.2 in Spack CI, the commit above had no effect (it's only requiring vtk-m > 2.1).

@cyrush per your comment, I’m tempted to restrict vtk-m to 2.1.0:2.1.99. It builds with 2.2, but hasn't been tested.

**Comment by adrienbernede** [comment]
@tldahlgren This PR is ready, I’d like you review on it (of another Spack team member).

**Comment by adrienbernede** [comment]
@adayton1 please note the migration of CARE package to CachedCMakePackage (based on implementation in radiuss-spack-configs)
@daboehme please note the migration of Caliper package to CachedCMakePackage (based on implementation in radiuss-spack-configs)
@rhornung67 please note the addition of RAJAPerf package (based on implementation in radiuss-spack-conifgs)

File: var/spack/repos/builtin/packages/raja-perf/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/raja-perf/package.py]: This is no longer a valid URL.

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/raja-perf/package.py]: The "closest" non-github URL may be https://software.llnl.gov/#RAJAPerf BUT you still have to search for `RAJAPerf` to narrow it down to the software.

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/raja-perf/package.py]: It isn't available from https://software.llnl.gov/radiuss/projects/, only `raja` yields the software "box" link.

File: var/spack/repos/builtin/packages/care/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/care/package.py]: It looks like the branch is actually `master`, not `main` (and there is no `main` branch ATM).

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Confirmed all affected version commits (or tags) *except* `care@master` (see comment).  There's also an issue with the URL for `raja-perf`.

FWIW. Some people are hesitant to change variant defaults since others *may* depend on them.

File: var/spack/repos/builtin/packages/raja-perf/package.py
Comment (rhornung67) [inline var/spack/repos/builtin/packages/raja-perf/package.py]:
```suggestion
    homepage = "https://github.com/LLNL/RAJAPerf"
```

File: var/spack/repos/builtin/packages/care/package.py
Comment (adrienbernede) [reply var/spack/repos/builtin/packages/care/package.py]: Done

File: var/spack/repos/builtin/packages/raja-perf/package.py
Comment (adrienbernede) [reply var/spack/repos/builtin/packages/raja-perf/package.py]: Done !

**Review by tldahlgren — DISMISSED** [review]
Affected homepages, branches, commits/tags LGTM.

File: var/spack/repos/builtin/packages/care/package.py
Comment (adrienbernede) [inline var/spack/repos/builtin/packages/care/package.py:74]: @adayton1 I didn't even notice this change I made. We had implicit_conversions set to "False" locally. Would you prefer to have it to "True" here in Spack ?
Thanks @tldahlgren for the warning.

File: var/spack/repos/builtin/packages/care/package.py
Comment (adrienbernede) [inline var/spack/repos/builtin/packages/care/package.py:79]: I set those to "False" which is coherent with `tests` being set to "False" itself.
Especially true for `benchmarks` which requires `tests`.
@adayton1 Are you OK with that ?

Comment (adrienbernede) [reply var/spack/repos/builtin/packages/care/package.py:79]: Please also validate "docs" below... maybe I shouldn't have changed that one...

File: var/spack/repos/builtin/packages/care/package.py
Comment (adayton1) [reply var/spack/repos/builtin/packages/care/package.py:74]: I think it should default to False.

File: var/spack/repos/builtin/packages/care/package.py
Comment (adayton1) [reply var/spack/repos/builtin/packages/care/package.py:79]: I'm OK with tests, benchmarks, examples, and docs all defaulting to False.

**Review by adayton1 — APPROVED** [review]
LGTM

**Comment by tldahlgren** [comment]
Is this PR ready to merge assuming all of the checks pass?

**Comment by adrienbernede** [comment]
@tldahlgren @v-dobrev I don't think the error with MFEM is caused by this pull request since it does not affect the MFEM package features nor any of MFEM dependencies involved in the build.
Instead, I think the MFEM build was triggered because the package was touched, but the error was introduced by another PR which hadn't triggered the MFEM builds.
```
  >> 590    ld.lld: error: undefined reference due to --no-allow-shlib-undefine
            d: dlclose
     591    >>> referenced by /opt/cray/pe/libsci/23.02.1.1/CRAY/9.0/x86_64/lib
            /libsci_cray.so
```

**Comment by adrienbernede** [comment]
> Is this PR ready to merge assuming all of the checks pass?

Yes it is.

**Comment by adrienbernede** [comment]
@tldahlgren @v-dobrev I reverted the changes in the MFEM package and applied them in a separate PR.

https://github.com/spack/spack/pull/46537

CI is now passing.

**Comment by adrienbernede** [comment]
@tldahlgren this is ready (only failure is coverage) ;)

**Comment by adrienbernede** [comment]
@tldahlgren I spotted an issue. Please do not merge.

**Comment by adrienbernede** [comment]
@trws, @tldahlgren being out of office this week, would you mind helping me getting this merged ?

**Comment by trws** [comment]
I'm just getting back from Australia so I'm not sure I'll be faster, but I can try.  What do you still need?

**Comment by adrienbernede** [comment]
@trws I need you (or someone) to hit the "approve" button. ;)

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**
